test/e2e/nodelatencymonitor_test.go

```diff
@@ -0,0 +1,194 @@
+package e2e
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	v1alpha1 "antrea.io/antrea/pkg/apis/crd/v1alpha1"
+	statsv1alpha1 "antrea.io/antrea/pkg/apis/stats/v1alpha1"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"k8s.io/apimachinery/pkg/api/errors"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+)
+
+const (
+	nodeLatencyMonitorName = "default"
+	nodeLatencyMonitorNS   = "kube-system"
+	pingIntervalSeconds    = 50
+)
+
+func TestNodeLatencyMonitor(t *testing.T) {
+	skipIfNotRequired(t)
+
+	data, err := setupTest(t)
+	if err != nil {
+		t.Fatalf("Failed to set up clients: %v", err)
+	}
+	defer teardownTest(t, data)
+
+	createOrUpdateNodeLatencyMonitorCR(t, data)
+
+	t.Run("testEnableLatencyProbes", func(t *testing.T) {
+		testEnableLatencyProbes(t, data)
+	})
+	t.Run("testRetrieveLatencyStats", func(t *testing.T) {
+		testRetrieveLatencyStats(t, data)
+	})
+	t.Run("testUpdatePingInterval", func(t *testing.T) {
+		testUpdatePingInterval(t, data)
+	})
+	t.Run("testDisableLatencyProbes", func(t *testing.T) {
+		testDisableLatencyProbes(t, data)
+	})
+}
+
+func createOrUpdateNodeLatencyMonitorCR(t *testing.T, data *TestData) {
+	nlm := &v1alpha1.NodeLatencyMonitor{
+		ObjectMeta: v1.ObjectMeta{
+			Name:      nodeLatencyMonitorName,
+			Namespace: nodeLatencyMonitorNS,
+		},
+		Spec: v1alpha1.NodeLatencyMonitorSpec{
+			PingIntervalSeconds: pingIntervalSeconds,
+		},
+	}
+
+	existingNLM, err := data.crdClient.CrdV1alpha1().NodeLatencyMonitors().Get(context.TODO(), nlm.Name, v1.GetOptions{})
+	if err != nil {
+		if errors.IsNotFound(err) {
+
+			_, err := data.crdClient.CrdV1alpha1().NodeLatencyMonitors().Create(context.TODO(), nlm, v1.CreateOptions{})
+			if err != nil {
+				t.Fatalf("Failed to create NodeLatencyMonitor CR: %v", err)
+			}
+		} else {
+
+			t.Fatalf("Failed to get NodeLatencyMonitor CR: %v", err)
+		}
+	} else {
+
+		nlm.ResourceVersion = existingNLM.ResourceVersion
+		_, err := data.crdClient.CrdV1alpha1().NodeLatencyMonitors().Update(context.TODO(), nlm, v1.UpdateOptions{})
+		if err != nil {
+			t.Fatalf("Failed to update NodeLatencyMonitor CR: %v", err)
+		}
+	}
+
+	waitForNodeLatencyMonitor(t, data)
+}
+
+func testEnableLatencyProbes(t *testing.T, data *TestData) {
+	waitForLatencyProbesEnabled(t, data)
+}
+
+func testDisableLatencyProbes(t *testing.T, data *TestData) {
+	err := data.crdClient.CrdV1alpha1().NodeLatencyMonitors().Delete(context.TODO(), nodeLatencyMonitorName, v1.DeleteOptions{})
+	if err != nil {
+		t.Fatalf("Failed to delete NodeLatencyMonitor CR: %v", err)
+	}
+
+	waitForLatencyProbesDisabled(t, data)
```

Comment:
does this really serve a purpose? The resource has no finalizers, so it should be deleted immediately, and I think there is no point in querying the API again to check for deletion. Is that no what you observed?